src/rpm/package.rs

```diff
@@ -770,7 +770,9 @@ impl PackageMetadata {
                         |mut acc, item| {
                             let (basename, dir_index) = item;
                             if let Some(dir) = dirs.get(dir_index as usize) {
-                                acc.push(PathBuf::from(dir).join(basename));
+                                let mut path = PathBuf::from(dir);
+                                path.push(basename);
```

Comment:
Is it possible to do `Path::from(dir).join(basename)` instead?  I believe that would also avoid the extra allocation, and perhaps more reliably.